### ISSUE TITLE
Add the GCNBMP model

### ISSUE THREAD

**Issue by cthoyt** [AstraZeneca/chemicalx#21]
- Please read the paper first. It is [here](https://www.sciencedirect.com/science/article/pii/S1046202320300608).
- There is also code-release with the paper [here](https://github.com/Minys233/GCN-BMP).
- After that read the contributing guidelines.
-   If there is an existing open-source version of the model please take a look.
-   ChemicalX is built on top of [PyTorch 1.10.](https://pytorch.org/) and [torchdrug](https://torchdrug.ai/).
- The library heavily builds on top on [torchdrug](https://torchdrug.ai/) and molecules in batches are PackedGraphs.
- There is already a model class under `./chemicalx/models/`
- Context features, drug level features, and labels are all FloatTensors.
- Look at the examples and tests under `./examples/` and `./tests/`.
- Add auxiliary layers as you see fit - please document these, add tests, and add these layers to the main readme.md if needed.
- Add typing to the initialization and forward pass.
- Non-data-dependent hyperparameters should have default values. 
- Please add tests under `./tests/` and make sure that your model/layer is tested with real data.
- Write an example under `./examples/`. What is the AUC on the test set? Is it reasonable?

**Comment by mughetto** [comment]
Hey @cthoyt 

I discussed with @benedekrozemberczki  and I'm gonna implement this one. I've had a look at the original code and the paper and I think the architecture is pretty clear for me now. Will let you know how it goes.

**Comment by cthoyt** [comment]
Excellent!
